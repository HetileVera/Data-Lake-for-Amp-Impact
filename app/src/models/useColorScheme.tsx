import { useTheme } from "@chakra-ui/react";

export function useColorScheme() {
  const colorScheme = useTheme().name;
  const tone = (value: number | undefined) => (value ? `${colorScheme}.${value}` : colorScheme);

  return { tone, colorScheme };
}
