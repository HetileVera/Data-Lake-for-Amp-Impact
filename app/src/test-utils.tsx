import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ChakraProvider } from "@chakra-ui/react";

import { theme } from "themes/orange";

export const CustomChakraProvider = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={theme}>
      <div data-testid="test-root">{children}</div>
    </ChakraProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, "wrapper">) => render(ui, { wrapper: CustomChakraProvider, ...options });

export * from "@testing-library/react";

export { customRender as render };
