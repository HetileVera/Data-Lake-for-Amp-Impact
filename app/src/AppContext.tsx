import React, { createContext, useContext } from "react";
import isEmpty from "lodash/isEmpty";
import throttle from "lodash/throttle";

import { AppStore, IAppStore } from "./models/AppStore";
import { IInstallation } from "./models/Installation";
import { isLocalDev, websiteUrlOrigin, assetWebsiteUrl, secretKey } from "./helpers/settings";
import { onSnapshot } from "mobx-state-tree";

// Create an instance of the root store (a.k.a appStore)
const appStore = AppStore.create({});

// Store state to local storage
const storeLocally = throttle(() => appStore.saveLocal(), 5000);

// Store state in S3 when applicable
const storeRemotely = throttle(() => appStore.saveRemote(), 30000);

// The initialization function for the root store
export function initializeStore() {
  if (!isEmpty(secretKey)) console.warn("Access key is bundled!");

  if (isLocalDev) {
    console.log(`isLocalDev '${isLocalDev}'`);
    console.log(`websiteUrlOrigin '${websiteUrlOrigin}'`);
    console.log(`assetWebsiteUrl '${assetWebsiteUrl}'`);

    if (window.location.host !== "localhost:3000") {
      console.warn("You should run the app on localhost!");

      throw new Error("App should run on localhost for development mode!");
    }
  }

  appStore.loadLocal();
  onSnapshot(appStore, (snapshot) => {
    storeLocally();
    storeRemotely();
  });
}

// Create the store context
export const StoreContext = createContext<IAppStore | undefined>(undefined);

// Create the store provider component
export const StoreProvider: React.FunctionComponent = ({ children }) => {
  return <StoreContext.Provider value={appStore}>{children}</StoreContext.Provider>;
};

// useStore hook
export function useStore(): IAppStore {
  const store = useContext(StoreContext);

  if (!store) {
    throw new Error("useStore() must be used within a StoreProvider");
  }

  return store;
}

// useInstallation hook
export function useInstallation(): IInstallation {
  const store = useStore();
  const installation = store.installation;

  if (!installation) {
    throw new Error("An installation instance was not created yet");
  }

  return installation;
}
