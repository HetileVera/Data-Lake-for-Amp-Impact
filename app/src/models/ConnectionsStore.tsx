import { useEffect } from "react";
import isEmpty from "lodash/isEmpty";
import { types, Instance, getParent } from "mobx-state-tree";

import { useInstallation } from "AppContext";
import { listAppflowConnections } from "api/list-appflow-connections";

import { BaseStore, isStoreNew, isStoreError, isStoreReady, isStoreLoading, isStoreReLoading } from "./BaseStore";
import { ICredentials } from "./helpers/Credentials";
import { IInstallation } from "./Installation";

/**
 * Represents the store for the AppFlow Salesforce connection names
 */
export const ConnectionsStore = BaseStore.named("ConnectionsStore")
  .props({
    connectionNames: types.array(types.string),
  })

  .views((self) => ({
    get empty() {
      return isEmpty(self.connectionNames);
    },

    get credentials(): ICredentials {
      const parent: IInstallation = getParent(self);
      return parent.credentials;
    },

    get region(): string {
      const parent: IInstallation = getParent(self);
      return parent.region;
    },
  }))

  .actions((self) => {
    const superReset = self.reset;
    return {
      reset() {
        superReset();
        self.connectionNames.clear();
      },

      async doLoad() {
        const accessKey = self.credentials.accessKey;
        const secretKey = self.credentials.secretKey;
        const region = self.region;

        const rawConnectionNames = await listAppflowConnections({ accessKey, secretKey, region });
        self.runInAction(() => {
          self.connectionNames.replace(rawConnectionNames);
        });
      },
    };
  });

// see https://mobx-state-tree.js.org/tips/typescript
export interface IConnectionsStore extends Instance<typeof ConnectionsStore> {}

export function useConnectionsStore() {
  const installation = useInstallation();
  const store = installation.connectionsStore;

  useEffect(() => {
    if (!isStoreNew(store)) return;
    store.load();
  }, [store]);

  return {
    isError: isStoreError(store),
    isReady: isStoreReady(store),
    isLoading: isStoreLoading(store),
    isReloading: isStoreReLoading(store),
    store,
  };
}
