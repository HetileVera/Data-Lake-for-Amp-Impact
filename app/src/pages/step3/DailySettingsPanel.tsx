import React from "react";
import { Box, Text, HStack, Divider, Alert, AlertIcon } from "@chakra-ui/react";
import { observer } from "mobx-react";

import { useInstallation } from "AppContext";
import { TimeInput } from "components/TimeInput";

export const DailySettingsPanel = observer(() => {
  const installation = useInstallation();
  const step = installation.importOptionsStep;
  const settings = step.dailySettings;
  const time = settings.time;

  const handleTimeChange = ({ hour, minute }: { hour: string; minute: string }): void => {
    time.setHour(hour);
    time.setMinute(minute);
  };

  return (
    <Box>
      <Alert status="warning" variant="left-accent" mt={0} mb={3} colorScheme="purple" alignItems="flex-end">
        <AlertIcon alignSelf="flex-start" />
        Daily import is not recommended if you have millions of records of data as they incur additional cost
      </Alert>

      <Box p={3} bg="purple.75" borderRadius="md" mb={3}>
        {settings.infoMessage}
      </Box>

      <Divider borderColor="purple.200" mb={3} />

      <HStack spacing={2} color="purple.600">
        <Text fontSize="sm" fontWeight="bold">
          Time
        </Text>
        <Text fontSize="xs">(24-hour format)</Text>
      </HStack>

      <TimeInput hour={time.hour} minute={time.minute} onChange={handleTimeChange} />
    </Box>
  );
});
