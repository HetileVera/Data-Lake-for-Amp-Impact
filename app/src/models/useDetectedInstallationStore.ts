import { isStoreError, isStoreLoading, isStoreReady, isStoreReLoading } from "models/BaseStore";

import { DetectedInstallationStore } from "./DetectedInstallationsStore";

// Create an instance of this store
const store = DetectedInstallationStore.create({});

export function useDetectedInstallationStore() {
  return {
    isError: isStoreError(store),
    isReady: isStoreReady(store),
    isLoading: isStoreLoading(store),
    isReloading: isStoreReLoading(store),
    store,
  };
}
