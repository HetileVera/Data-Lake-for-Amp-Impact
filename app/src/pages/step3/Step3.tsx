import React from "react";
import { Box, HStack, Button, Divider } from "@chakra-ui/react";
import { ArrowForwardIcon, ArrowBackIcon } from "@chakra-ui/icons";
import { observer } from "mobx-react";
import { useNavigate } from "react-router-dom";

import { useInstallation } from "AppContext";
import { Layout } from "components/Layout";
import { ImportFrequencyPanel } from "pages/step3/ImportFrequencyPanel";
import { useMetadataStore } from "models/MetadataStore";
import { useColorScheme } from "models/useColorScheme";

import { StoreStatusInfo } from "pages/step3/StoreStatusInfo";
import { DataModelPanel } from "pages/step3/DataModelPanel";

export const Step3 = observer(() => {
  const { tone, colorScheme } = useColorScheme();
  const installation = useInstallation();
  const step = installation.importOptionsStep;
  const navigate = useNavigate();
  const [isSubmitting, setSubmitting] = React.useState(false);
  const { isReady, store } = useMetadataStore();
  const selectedObjectsCount = store.selectedObjects.length;

  const handlePrevious = () => {
    navigate("/steps/2");
  };

  const handleSubmit: React.MouseEventHandler<HTMLButtonElement> = async () => {
    step.markCompleted();
    setSubmitting(false);
    navigate("/steps/4");
  };

  React.useEffect(() => {
    step.markStarted();
    window.scrollTo(0, 0); // TODO - bring me back
  }, [step]);

  const canProceed = isReady && selectedObjectsCount > 0;

  return (
    <Layout
      step={3}
      title="Select data import options"
      description="In this step, you can pick the standard Amp Impact import options for your Salesforce data or customize these options to fit your needs. You also get to specify the schedule at which data is imported."
    >
      <Box borderRadius="lg" boxShadow="base" bg={tone(25)} mt={4} p={8}>
        <StoreStatusInfo />
        <ImportFrequencyPanel />
        {isReady && <Divider borderColor={tone(100)} pt={8} mb={6} />}
        <DataModelPanel />
      </Box>

      <HStack justifyContent="space-between" p={3} pt={6} mb={12}>
        <Button
          colorScheme={colorScheme}
          size="md"
          _hover={{ bg: tone(100) }}
          leftIcon={<ArrowBackIcon />}
          disabled={isSubmitting}
          variant="ghost"
          onClick={handlePrevious}
        >
          Previous
        </Button>
        <Button
          id="step3-btn-next"
          colorScheme={colorScheme}
          size="md"
          loadingText="Processing"
          rightIcon={<ArrowForwardIcon />}
          ml={3}
          isLoading={isSubmitting}
          disabled={!canProceed || isSubmitting}
          onClick={handleSubmit}
        >
          Next
        </Button>
      </HStack>
    </Layout>
  );
});
