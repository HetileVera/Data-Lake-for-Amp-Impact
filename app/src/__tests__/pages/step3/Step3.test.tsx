import { cleanup, screen } from "@testing-library/react";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import { render, RenderWithChakra } from "test-utils";

import * as appContext from "AppContext";
import * as metadataStore from "models/MetadataStore";
import { Step3 } from "pages/step3/Step3";

describe("Step2 component", () => {
  afterEach(cleanup);

  test("snapshot", () => {
    // @ts-ignore
    jest.spyOn(appContext, "useInstallation").mockImplementation(() => {
      return {
        id: "1",
        startDate: "2022-01-01",
        startedBy: "test-user",
        accountId: "test-account-id",
        region: "us-east-1",
        appFlowConnectionName: "test-appflow-connection-name",
        credentials: [],
        deploymentStep: "1",
        reviewStep: "1",
        nextStepNumber: "2",
        importOptionsStep: {
          markStarted: jest.fn(),
          markCompleted: jest.fn(),
        },
      };
    });
    // @ts-ignore
    jest.spyOn(metadataStore, "useMetadataStore").mockImplementation(() => {
      return {
        isReady: true,
        store: {
          listAll: [{}],
          selectedObjects: [{}],
          missingObjects: [],
          excludedObjects: [],
        },
      };
    });

    const tree = create(
      <RenderWithChakra>
        <appContext.StoreProvider>
          <BrowserRouter>
            <Step3 />
          </BrowserRouter>
        </appContext.StoreProvider>
      </RenderWithChakra>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("render properly", () => {
    // @ts-ignore
    jest.spyOn(appContext, "useInstallation").mockImplementation(() => {
      return {
        id: "1",
        startDate: "2022-01-01",
        startedBy: "test-user",
        accountId: "test-account-id",
        region: "us-east-1",
        appFlowConnectionName: "test-appflow-connection-name",
        credentials: [],
        deploymentStep: "1",
        reviewStep: "1",
        nextStepNumber: "2",
        importOptionsStep: {
          markStarted: jest.fn(),
          markCompleted: jest.fn(),
        },
      };
    });
    // @ts-ignore
    jest.spyOn(metadataStore, "useMetadataStore").mockImplementation(() => {
      return {
        isReady: true,
        store: {
          listAll: [{}],
          selectedObjects: [{}],
          missingObjects: [],
          excludedObjects: [],
        },
      };
    });

    render(
      <appContext.StoreProvider>
        <BrowserRouter>
          <Step3 />
        </BrowserRouter>
      </appContext.StoreProvider>
    );

    expect(screen.getAllByText(/select data import options/i)).toBeTruthy();
    expect(screen.getByText(/you can pick the standard npsp import options/i)).toBeInTheDocument();
    expect(screen.getByText("Previous")).toBeInTheDocument();
    expect(screen.getByText("Next")).toBeInTheDocument();
  });
});
