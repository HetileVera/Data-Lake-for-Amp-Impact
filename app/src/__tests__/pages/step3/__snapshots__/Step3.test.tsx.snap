// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step3 component snapshot 1`] = `
Array [
  <div
    className="css-1v55i00"
    data-testid="curved-box"
  />,
  <div
    className="chakra-container css-1szj4qk"
  >
    <div
      className="chakra-stack css-olw0lh"
    >
      <div
        className="chakra-stack css-18lyn6q"
      >
        <div
          className="css-1i3fafl"
        >
          <img
            className="chakra-image css-0"
            src="Salesforce_Corporate_Logo_RGB.png"
          />
        </div>
        <div
          className="chakra-stack__divider css-r2vkku"
        />
        <div
          className="css-nutmff"
        >
          <img
            className="chakra-image css-ir5tdx"
            src="AWS_logo_RGB_BLK.png"
          />
        </div>
      </div>
      <div
        className="chakra-stack css-1n7gsjr"
      >
        <div
          className="css-18n4g2t"
        >
          <h2
            className="chakra-heading css-1opsxio"
          >
            Datalake for Nonprofits, Powered by AWS
          </h2>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="css-79elbk"
  >
    <div
      className="css-t7j6xp"
      data-testid="curved-box"
    />
    <div
      className="css-15dboc9"
    >
      <div
        className="chakra-container css-h66qep"
      >
        <div
          className="chakra-stack css-84zodg"
        >
          <span
            className="css-9xsoz7"
          >
            1
          </span>
          <span
            className="css-9xsoz7"
          >
            2
          </span>
          <span
            className="css-hfqsm"
          >
            3
          </span>
          <span
            className="css-ulmpab"
          >
            4
          </span>
          <span
            className="css-ulmpab"
          >
            5
          </span>
          <span
            className="css-ulmpab"
          >
            6
          </span>
        </div>
      </div>
    </div>
    <div
      className="chakra-container css-11yz6e2"
    >
      <h2
        className="chakra-heading css-noy8un"
      >
        Select data import options
      </h2>
      <div
        className="css-19a4hof"
      >
        <div
          className="chakra-stack css-11wh0gm"
        >
          <div
            className="css-1y0bycm"
          >
            In this step, you can pick the standard NPSP import options for your Salesforce data or customize these options to fit your needs. You also get to specify the schedule at which data is imported.
          </div>
        </div>
      </div>
      <div
        className="css-1cxlklx"
      >
        <div
          className="css-1ui15d5"
        >
          <div
            className="css-1lpnqa0"
          >
            Retrieving your Salesforce data schema, this might take a few minutes
            <br />
            NaN
             %
          </div>
          <div
            className="css-1jer4t"
          >
            <div
              aria-valuemax={100}
              aria-valuemin={0}
              className="css-1585z0b"
              data-indeterminate=""
              role="progressbar"
              style={
                Object {
                  "width": "0%",
                }
              }
            />
          </div>
        </div>
        <div
          className="css-1frymol"
        >
          <h2
            className="chakra-heading css-8297jb"
          >
            How often should we import the data?
          </h2>
          <div
            className="chakra-stack css-flp588"
          >
            <div
              className="css-2m5dz5"
            >
              <div
                className="chakra-radio-group css-0"
                role="radiogroup"
              >
                <div
                  className="css-1m6qo81"
                  onClick={[Function]}
                >
                  <div
                    className="css-1jvs9h2"
                  >
                    <label
                      className="chakra-radio css-1pw4d56"
                    >
                      <input
                        checked={false}
                        className="chakra-radio__input"
                        id="radio-1"
                        name="radio-1"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        style={
                          Object {
                            "border": "0px",
                            "clip": "rect(0px, 0px, 0px, 0px)",
                            "height": "1px",
                            "margin": "-1px",
                            "overflow": "hidden",
                            "padding": "0px",
                            "position": "absolute",
                            "whiteSpace": "nowrap",
                            "width": "1px",
                          }
                        }
                        type="radio"
                      />
                      <span
                        aria-hidden={true}
                        className="chakra-radio__control css-ww8qvl"
                        onMouseDown={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                      />
                      <span
                        className="chakra-radio__label css-1i66d7g"
                        onMouseDown={[Function]}
                        onTouchStart={[Function]}
                      >
                        Monthly
                      </span>
                    </label>
                    <div
                      className="css-0"
                    >
                      <span
                        className="chakra-badge css-1690ih9"
                      >
                        Recommended
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  className="css-1rj9xgn"
                  onClick={[Function]}
                >
                  <label
                    className="chakra-radio css-1pw4d56"
                  >
                    <input
                      checked={false}
                      className="chakra-radio__input"
                      id="radio-1"
                      name="radio-1"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      style={
                        Object {
                          "border": "0px",
                          "clip": "rect(0px, 0px, 0px, 0px)",
                          "height": "1px",
                          "margin": "-1px",
                          "overflow": "hidden",
                          "padding": "0px",
                          "position": "absolute",
                          "whiteSpace": "nowrap",
                          "width": "1px",
                        }
                      }
                      type="radio"
                    />
                    <span
                      aria-hidden={true}
                      className="chakra-radio__control css-ww8qvl"
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                    />
                    <span
                      className="chakra-radio__label css-1i66d7g"
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    >
                      Weekly
                    </span>
                  </label>
                </div>
                <div
                  className="css-1rj9xgn"
                  onClick={[Function]}
                >
                  <label
                    className="chakra-radio css-1pw4d56"
                  >
                    <input
                      checked={false}
                      className="chakra-radio__input"
                      id="radio-1"
                      name="radio-1"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      style={
                        Object {
                          "border": "0px",
                          "clip": "rect(0px, 0px, 0px, 0px)",
                          "height": "1px",
                          "margin": "-1px",
                          "overflow": "hidden",
                          "padding": "0px",
                          "position": "absolute",
                          "whiteSpace": "nowrap",
                          "width": "1px",
                        }
                      }
                      type="radio"
                    />
                    <span
                      aria-hidden={true}
                      className="chakra-radio__control css-ww8qvl"
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                    />
                    <span
                      className="chakra-radio__label css-1i66d7g"
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    >
                      Daily
                    </span>
                  </label>
                </div>
              </div>
            </div>
            <div
              className="chakra-stack__divider css-1gvtail"
            />
            <div
              className="css-1dku7pm"
            />
          </div>
        </div>
        <hr
          aria-orientation="horizontal"
          className="chakra-divider css-zp35to"
        />
        <div
          className="css-1wrpr7j"
        >
          <h2
            className="chakra-heading css-1pprtor"
          >
            Salesforce NPSP data model
          </h2>
          <div
            className="chakra-alert css-1y8tq1o"
            role="alert"
          >
            <div
              className="css-0"
            >
              <div
                className="chakra-alert__title css-1vqyp4x"
              >
                Objects to import
                <span
                  className="chakra-badge css-ni2gbt"
                >
                  1
                </span>
              </div>
              <div
                className="chakra-alert__desc css-1o140vv"
              >
                If needed, you can include or exclude 
                <b>
                  objects
                </b>
                 by clicking on the object name.
                <p
                  className="chakra-text css-1art13b"
                >
                  You can also exclude 
                  <b>
                    fields
                  </b>
                   by clicking on the 'Customize Fields' button.
                </p>
              </div>
            </div>
          </div>
          <div
            className="css-11ycgl2"
          >
            <div
              className="css-eypmdn"
            >
              <div
                className="chakra-radio-group css-15o7o8w"
                role="radiogroup"
              >
                <div
                  className="css-gdii58"
                >
                  <div
                    className="css-1nda4v5"
                  >
                    <p
                      className="chakra-text css-yii78z"
                    >
                      Show
                    </p>
                    <label
                      className="chakra-radio css-fjnot9"
                      data-checked=""
                    >
                      <input
                        checked={true}
                        className="chakra-radio__input"
                        disabled={false}
                        id="radio-1"
                        name="radio-1"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        style={
                          Object {
                            "border": "0px",
                            "clip": "rect(0px, 0px, 0px, 0px)",
                            "height": "1px",
                            "margin": "-1px",
                            "overflow": "hidden",
                            "padding": "0px",
                            "position": "absolute",
                            "whiteSpace": "nowrap",
                            "width": "1px",
                          }
                        }
                        type="radio"
                        value="1"
                      />
                      <span
                        aria-hidden={true}
                        className="chakra-radio__control css-1taemt2"
                        data-checked=""
                        onMouseDown={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                      />
                      <span
                        className="chakra-radio__label css-1i66d7g"
                        data-checked=""
                        onMouseDown={[Function]}
                        onTouchStart={[Function]}
                      >
                        <p
                          className="chakra-text css-surgn5"
                        >
                          All Objects
                          <span
                            className="chakra-badge css-7hgm7x"
                          >
                            1
                          </span>
                        </p>
                      </span>
                    </label>
                  </div>
                  <label
                    className="chakra-radio css-1pw4d56"
                  >
                    <input
                      checked={false}
                      className="chakra-radio__input"
                      disabled={false}
                      id="radio-1"
                      name="radio-1"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      style={
                        Object {
                          "border": "0px",
                          "clip": "rect(0px, 0px, 0px, 0px)",
                          "height": "1px",
                          "margin": "-1px",
                          "overflow": "hidden",
                          "padding": "0px",
                          "position": "absolute",
                          "whiteSpace": "nowrap",
                          "width": "1px",
                        }
                      }
                      type="radio"
                      value="2"
                    />
                    <span
                      aria-hidden={true}
                      className="chakra-radio__control css-1taemt2"
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                    />
                    <span
                      className="chakra-radio__label css-1i66d7g"
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    >
                      <p
                        className="chakra-text css-surgn5"
                      >
                        Included Objects
                        <span
                          className="chakra-badge css-7hgm7x"
                        >
                          1
                        </span>
                      </p>
                    </span>
                  </label>
                  <label
                    className="chakra-radio css-1pw4d56"
                  >
                    <input
                      checked={false}
                      className="chakra-radio__input"
                      disabled={false}
                      id="radio-1"
                      name="radio-1"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      style={
                        Object {
                          "border": "0px",
                          "clip": "rect(0px, 0px, 0px, 0px)",
                          "height": "1px",
                          "margin": "-1px",
                          "overflow": "hidden",
                          "padding": "0px",
                          "position": "absolute",
                          "whiteSpace": "nowrap",
                          "width": "1px",
                        }
                      }
                      type="radio"
                      value="3"
                    />
                    <span
                      aria-hidden={true}
                      className="chakra-radio__control css-1taemt2"
                      onMouseDown={[Function]}
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                    />
                    <span
                      className="chakra-radio__label css-1i66d7g"
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                    >
                      <p
                        className="chakra-text css-gkdncc"
                      >
                        Excluded Objects
                        <span
                          className="chakra-badge css-7hgm7x"
                        >
                          0
                        </span>
                      </p>
                    </span>
                  </label>
                </div>
              </div>
              <table
                className="chakra-table css-1ijrwz3"
                role="table"
              >
                <thead
                  className="css-0"
                >
                  <tr
                    className="css-0"
                    role="row"
                  >
                    <th
                      className="css-e8a57g"
                    >
                      Objects
                    </th>
                    <th
                      className="css-at1ona"
                    >
                      Fields
                    </th>
                    <th
                      className="css-1fsrd9l"
                      data-is-numeric={true}
                    />
                  </tr>
                </thead>
                <tbody
                  className="css-0"
                >
                  <tr
                    className="css-4g6ai3"
                    role="row"
                  >
                    <td
                      className="css-1luj9dq"
                      colSpan={1}
                      role="gridcell"
                    >
                      <div
                        className="css-k008qs"
                      >
                        <svg
                          className="chakra-icon css-elgig9"
                          fill="currentColor"
                          focusable={false}
                          height="1em"
                          onClick={[Function]}
                          stroke="currentColor"
                          strokeWidth="0"
                          style={
                            Object {
                              "color": undefined,
                            }
                          }
                          viewBox="0 0 16 16"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
                          />
                        </svg>
                        <div
                          className="css-1nx7hf7"
                          onClick={[Function]}
                        >
                          <div
                            className="css-11cuipc"
                          />
                          <div
                            className="css-nrq09u"
                          />
                        </div>
                      </div>
                    </td>
                    <td
                      className="css-1luj9dq"
                      data-is-numeric={true}
                      role="gridcell"
                    />
                    <td
                      className="css-j803ed"
                      role="gridcell"
                    />
                  </tr>
                </tbody>
              </table>
              <div
                className="css-qzs98u"
              >
                <div
                  className="css-1p8q2i6"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="chakra-stack css-1aoc1cb"
      >
        <button
          className="chakra-button css-a63yqb"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            className="chakra-button__icon css-1wh2kri"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                fill="currentColor"
              />
            </svg>
          </span>
          Previous
        </button>
        <button
          className="chakra-button css-zb27lh"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Next
          <span
            className="chakra-button__icon css-1hzyiq5"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>,
  <span />,
]
`;
